### PR TITLE
SSP: Use *BIT* instead of raw raw bit operations

### PR DESCRIPTION
**Pull request by ktrzcinx**
Introduced solution is more elegant and reliable because of
masking given argument in SET_BITS.
Moreover remove unused masks which usability is taken by SET_BITS
macro with given bits range.

Upgrade BIT macro to use 1UL instead of 1

Move conversion from SSCR3 value to RFL to bit description section

### CODE REVIEW

File: src/include/sof/drivers/ssp.h [hunk 1]
```diff
@@ -154,6 +154,12 @@ extern const struct dai_driver ssp_driver;
 #define SSP_REG_MAX	SSIOC
 #endif
 
+/* SSTSA bits */
+#define SSTSA_TXEN		BIT(8)
+
+/* SSRSA bits */
+#define SSRSA_RXEN		BIT(8)
```
Comment (paulstelian97) [inline src/include/sof/drivers/ssp.h]: I assume there are no other bits? I usually put even those that I won't really use in the driver if the hardware offers them but I'll leave this up to you to decide.

Comment (ktrzcinx) [reply src/include/sof/drivers/ssp.h]: Now there are all

Comment (paulstelian97) [reply src/include/sof/drivers/ssp.h]: Perfect :)

**Comment by lgirdwood** [comment]
SOFCI TEST

File: src/include/sof/bit.h [hunk 2]
```diff
@@ -8,7 +8,12 @@
 #ifndef __SOF_BIT_H__
 #define __SOF_BIT_H__
 
+#if ASSEMBLY
 #define BIT(b)			(1 << (b))
```
Comment (jajanusz) [inline src/include/sof/bit.h]: I'm suprised it works in our asm. not Every asm supports it.

Comment (jajanusz) [reply src/include/sof/bit.h]: @lgirdwood I think that in kernel macros like this are just for C, how do they deal with ASM?

Comment (lgirdwood) [reply src/include/sof/bit.h]: This should also work for all pre-processed assembly as BIT() will just be turned into a number.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
Lets rerun CI, I'm gearing issues with some DUTs

**Comment by ktrzcinx** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@ktrzcinx BDW CI is back now.

**Comment by ktrzcinx** [comment]
Great

File: src/include/sof/bit.h (hunk 2 above)
Comment (jajanusz) [reply src/include/sof/bit.h]: @lgirdwood For xtensa only `(1 << x)` works, `(1U << x)`, is not, so I guess it's something asm-compiler-specific.

**Comment by lgirdwood** [comment]
@ktrzcinx ok, BDW is good but we are seeing a DMA timeout on BYT
```
  0      1         WAIT         141209994.635417 141210000.000000       src/lib/wait.c:39  	ewt
    0      1          DMA         141210001.927083         7.291667 src/drivers/dw/dma.c:413 	dw_dma_stop() error: dma 1 channel 1 timeout
    0      1           SA         141210015.312500        13.385417      src/lib/agent.c:56  	validate(), ll drift detected, delta = 27639
```
Channel is timing out on stop drain.

**Comment by lgirdwood** [comment]
@ktrzcinx lets split this into 4 PRs and let CI bisect. Fwiw, we do have some DW DMA bugs that may impact this, see #2419 . One of these PRs could also be "fixing" precedence or could be changing a register bit.

**Comment by ktrzcinx** [comment]
@lgirdwood changes introduced in this PR shouldn't change fw functionality at all, only in commit "ssp: Use *BIT* macros instead raw bit operations" I change something like `#define foo(x) (x << 3)` to `#define foo(x) SET_BITS(5, 4, x)` according to datasheet and it should be the only one functional change. But it split changes to 4 PR as you wish and lest wait for CI results.

**Comment by ktrzcinx** [comment]
Splitted into #2446 #2447 #2448 #2449

**Comment by lgirdwood** [comment]
@ktrzcinx great thanks, lets close this one and let CI do it's work.

**Pull request closed without merging**